web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs

```diff
@@ -0,0 +1,345 @@
+                        // info!("Send! {:?}", stat_sender);
+                    } else {
+                        panic!("Stat sender was not spawned!");
+                    }
+
+                    // Create a new stats object
+                    // Add it to the StatBuffer
+                }
+
+                // Let the stat_sender spawn / flush ...
+                // spawned_app.app.stat_sender.aggregate_and_save_loop()
+                // Send a signal to save ...
+            }
+
+            // (3) Await that all items are properly processed
+            // TODO: Await all the background handles
+
+            // Only after this mark all the items as processed / completed
+
+            // If the items are in rpc_v2, delete the initial items from the database
+
+            // return Ok(());
+
+            // (4) Update the batch in the old table with the current timestamp (Mark the batch as migrated)
+            let old_record_ids = old_records.iter().map(|x| x.id);
+            let update_result: UpdateResult = rpc_accounting::Entity::update_many()
+                .col_expr(
+                    rpc_accounting::Column::Migrated,
+                    Expr::value(Value::ChronoDateTimeUtc(Some(Box::new(
+                        chrono::offset::Utc::now(),
+                    )))),
+                )
+                .filter(rpc_accounting::Column::Id.is_in(old_record_ids))
+                // .set(pear)
+                .exec(db_conn)
+                .await?;
+
+            info!("Update result is: {:?}", update_result);
+
+            // (N-1) Mark the batch as migrated
+            // break;
+        }
+
+        info!(
+            "Background handles (2) are: {:?}",
+            important_background_handles
+        );
+
+        // Drop the handle
+        // Send the shutdown signal here (?)
+        // important_background_handles.clear();
+
+        // Finally also send a shutdown signal
+
+        drop(stat_sender);
+        // match app_shutdown_sender.send(()) {
+        //     Err(x) => {
+        //         panic!("Could not send shutdown signal! {:?}", x);
+        //     }
+        //     _ => {}
+        // };
+
+        // TODO: Should we also write a short verifier if the migration was successful (?)
+
+        // Drop the background handle, wait for any tasks that are on-going
+        while let Some(x) = important_background_handles.next().await {
+            info!("Returned item is: {:?}", x);
+            match x {
+                Err(e) => {
+                    error!("{:?}", e);
+                }
+                Ok(Err(e)) => {
+                    error!("{:?}", e);
+                }
+                Ok(Ok(_)) => {
+                    // TODO: how can we know which handle exited?
+                    info!("a background handle exited");
+                    // Pop it in this case?
+                    continue;
+                }
+            }
+        }
```

Comment:
i'm pretty sure this should be higher up (at step 3).

Reply:
so we basically re-spawn (and delete) the sender at each batch?

Reply:
anything above will make it respawn at each batch as I understand, see https://github.com/yenicelik/web3-proxy/blob/stats_v3/web3_proxy/src/bin/web3_proxy_cli/migrate_stats_to_v2.rs#L274